### PR TITLE
Added comments and Flake8

### PR DESCRIPTION
**Pull request by AlexanderBlake**
## Describe your changes
This Pull Request adds comments, ensures the program passes Flake8's lint rules, and adds Flake8 for all PR's and pushes.

## I have tested my code
- [X] Yes
- [ ] No. Reason: 

## Did you update the README
- [ ] Yes
- [X] No. Reason: This PR does not modify the design of the project.

### CODE REVIEW

File: .github/workflows/action.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: flake8 Lint
+
+on: [push, pull_request]
+
+jobs:
+  flake8-lint:
+    runs-on: ubuntu-latest
```
Comment (B1gbuddy) [inline .github/workflows/action.yaml:7]: what about windows?

**Review by nickhouses — COMMENTED** [review]
LGTM!

**Review by B1gbuddy — APPROVED** [review]
Do we have to use windows too?

File: .github/workflows/action.yaml (hunk 1 above)
Comment (AlexanderBlake) [reply .github/workflows/action.yaml:7]: This is for flake8 running on Github. It needs to be the latest version of Ubuntu.

**Pull request merged**